src/parser.rs

```diff
@@ -1,9 +1,18 @@
+use plc_ast::{
+    ast::{
+        AccessModifier, ArgumentProperty, AstStatement, CompilationUnit, DataType, DataTypeDeclaration,
+        DirectAccessType, GenericBinding, HardwareAccessType, Implementation, LinkageType, NewLines,
+        PolymorphismMode, Pou, PouType, SourceRange, SourceRangeFactory, TypeNature, UserTypeDeclaration,
+        Variable, VariableBlock, VariableBlockType,
+    },
+    id_provider::IdProvider,
+};
+
 // Copyright (c) 2020 Ghaith Hachem and Mathias Rieder
 use crate::{
-    ast::*,
     diagnostics::Diagnostician,
     expect_token,
-    lexer::{self, IdProvider},
+    lexer::{self},
```

Comment:
I know this is extremely nitpicky, but maybe you can change this to `lexer::{self, ParseSession, Token, Token::*},`?